src/dft.cpp

```diff
@@ -174,7 +174,30 @@ dft_chunk *fields::add_dft(component c, const volume &where, const double *freq,
   dft_chunk_data data;
   data.c = c;
   data.vc = vc;
+
+  if (decimation_factor == 0) {
+    double tol = 1e-7;
+    double src_freq_max = 0;
+    for (src_time *s = sources; s; s = s->next) {
+      if (s->get_fwidth(tol) == 0) {
+          decimation_factor = 1;
+        } else {
+        if (s->frequency().real()+0.5*s->get_fwidth(tol) > src_freq_max) {
+          src_freq_max = s->frequency().real()+0.5*s->get_fwidth(tol);
+        }
+      }
+    }
+    double freq_max = 0;
+    for (size_t i = 0; i < Nfreq; ++i) {
+      if (freq[i] > freq_max)
+        freq_max = freq[i];
+    }
+    if ((freq_max > 0) && (src_freq_max > 0)) {
+      decimation_factor = 1/(dt*(freq_max + src_freq_max));
```

Comment:
Probably should be
```cxx
decimation_factor = int(std::ceil(1/(dt*(freq_max + src_freq_max))));
```
(some compilers complain if you don't do an explicit `int` cast, and we should use `ceil` rather than the default truncation to be safe)

Reply:
Fixed.